### DISCUSSION TITLE
v4.44.0

### DISCUSSION THREAD

**Discussion by xiaokangwang** [v2fly/v2ray-core#1448] [General]
This release includes security enhancement for all users.

## !!! Important SECURITY enhancement !!!
* Fix DoS attack vulnerability in CommandSwitchAccountFactory. (Thanks   @geeknik)

## Fix 
* Apply timeout to DNS outbound. (#1330 Thanks @nekohasekai)

## Security Advisory
This update fixes a DoS vulnerability in V2Ray. This vulnerability allows a VMess Server controlled by an attacker to crash a VMess Client by sending a specially crafted handshake response reply with an (optional) VMess SwitchAccount Command that is one byte shorter than expected. This vulnerability does NOT allow the attacker to retrieve any information from a client other than it used an unpatched version of the software and does NOT attacker to control the unpatched software or system. It is strongly recommended for all users to apply this security update at the earliest possible opportunity. We would like to thank @geeknik for the responsible disclosure of this vulnerability.

此更新修复了在 V2Ray 中的一个拒绝服务攻击漏洞。这个漏洞允许攻击者控制的 VMess 服务器迫使 VMess 客户端崩溃。这个漏洞可以通过在 VMess 握手阶段向客户端发送一个恶意的回复数据包被触发，触发漏洞数据包的内容是比正确内容少一个字节的 VMess 切换账户指令。 攻击者 **无法** 通过这个漏洞获取来自客户端任何信息（除客户端尚未应用此安全更新以外），也 **不会** 允许攻击者控制客户端软件或系统。强烈推荐所有用户在第一时间应用本安全更新。我们在此感谢 @geeknik 将此漏洞负责任的披露给我们。

## Important Message 

V2Ray(V2Fly) will pre-release its next major version: V2Ray V5. In addition to functionality improvements, it will include a new configuration format and infrastructure changes that streamline the development of new protocols and functionalities.

The V4 version of the V2Ray will then enter maintenance mode. No additional features will be added by core developers, while contributors may still send pull requests for new features. It will receive bug fixes and security updates from core developers for a limited period until the full release of the V5 version. Contributors are encouraged to fork and base their changes from V5 version branch, instead of V4 version to reduce merge conflict. If you have already started the development of a change based on V4 version, you may send your pull request to V4 branch for a limited period. The core developers will cherry-pick that change into V5 version on your behave.

The pre-release version of V2Ray V5 may still contain bugs or inconsistencies. Some breaking updates to it are expected. You will need to change your configuration or codebase alongside us if you switch to V2Ray V5 now.

When V2Ray V5 pre-releases the master branch will switch to V5 version of the codebase, and the new V5 binary will be pre-released in the Github Release. They are not suitable for an automatic update from the V4 version.

## Notice

Due to increase in size of the `geoip.dat` file recently, devices with insufficient ROM/RAM are experiencing difficulties in using V2Ray. The solution is as follows:

* For **RAM** insufficient devices: Enable the Geodata loader optimized for memory-constrained devices by setting the environment variable `V2RAY_CONF_GEOLOADER` to value `memconservative`. For more details, see [documentation](https://www.v2fly.org/config/env.html#geodata-%E6%96%87%E4%BB%B6%E5%8A%A0%E8%BD%BD%E5%99%A8).
* For **ROM** insufficient devices:
  * Use the newly added GeoIP file `geoip-only-cn-private.dat` in the zip package or download it from [release page](https://github.com/v2fly/geoip/releases), which only contains GeoIP list `geoip:cn` and `geoip:private`, or
  * Customize your own GeoIP file via project [v2fly/geoip](https://github.com/v2fly/geoip).

<hr /><em>This discussion was created from the release <a href='https://github.com/v2fly/v2ray-core/releases/tag/v4.44.0'>v4.44.0</a>.</em>

**Comment by daryl000** [comment]
太好了~ 被这个DOS攻击困扰已久~

另外：
"攻击者 无法 通过这个漏洞获取来自客户端任何信息（除客户端尚未应用此安全更新以外）"
这句话的意思是在没有更新4.44.0之前，客户端的信息是会被攻击者获取的吗？如果信息会被获取，都包括哪些信息( PATH?, UUID?, Server IP? 域名？或者甚至是全部的服务器信息？)
更新之后，是否需要将V2RAY配置重置并更换，以保证使用安全？

谢谢！

**Reply by hebo6** [reply]
兄弟，“除客户端尚未应用此安全更新以外” 修饰的是 “任何信息”。
除客户端尚未应用此安全更新以外的 任何信息 都无法被攻击者 获取。

**Reply by daryl000** [reply]
更新4.44后依然会被DOS攻击导致V2RAY崩溃。
LOG例如
wan-ip:port accepted lan-ip:port
每秒高达800+次，V2RAY坚持不了几分钟就DOWN了。
请问如何解决。谢谢了

**Reply by xiaokangwang** [reply]
修复的不是你描述的这个问题，感觉你这个是被扫描了。如果你不需要从互联网访问的话，建议客户端只监听在环回设备上如 `127.0.0.1` .

**Reply by daryl000** [reply]
感谢作者回复。
不需要互联网访问，我的V2RAY客户端监就是127.0.0.1。这种扫描几乎每24小时发生1-2次。想请问如何解决这个问题吗？iptables可以解决这个问题吗？这个问题已经困扰我很久，技术有限，一直都没有找到正确的对应办法。THANKS

看到了V5版本的release，
‘Burst Observatory Measure connection quality to a significant amount of outbound at the same time. Thanks @qjebbs.’
请问这个功能跟我的问题有关吗？可以解决吗？

**Reply by xiaokangwang** [reply]
我还是不太清除具体是为什么炸的，如果监听的是127.0.0.1的话，应该不会接收来自wan-ip的流量吧。。。最好抓包一下确定问题的原因是什么。
Burst Observatory 和这个问题关系不大。你这个情况感觉是被别人滥用了。

**Reply by daryl000** [reply]
抓包并不会。我可以去现学一下，等到再次出现上述情况的时候，尝试抓包看一下。哦SORRY，最上面的情况我写错了。出现的问题是：
wan-ip:port(e.g. 123.123.112.32:45589) ACCEPTED wan-ip:1088 (iptables设置的透明代理端口)
抱歉我写错了。

**Comment by CyberShadowX** [comment]
macOS 运行QV2Ray ， 提示内核加载失败，失败代码：2
这个问题怎么解决啊
